### PR TITLE
[SALAD-22778] WebApp: Demand monitor page connect

### PR DESCRIPTION
**Pull request by vitto-moz**
image

### CODE REVIEW

File: packages/web-app/src/modules/demand-monitor-views/DemandMonitorPage/DemandMonitorTable/DemandMonitorTable.tsx [hunk 1]
```diff
@@ -76,74 +101,135 @@ const styles: (theme: SaladTheme) => Record<string, CSS.Properties> = (theme: Sa
...
+    if (sortOrder.columnKey === columnKey) {
+      setSortOrder({ columnKey, sorted: sortOrder.sorted === 'descending' ? 'ascending' : 'descending' })
+      return
+    }
+
+    setSortOrder({ columnKey, sorted: 'descending' })
+  }
+
+  let sortedDemandedHardwarePerformanceList = sortHardwareDemandPerformance(
+    toJS(demandedHardwarePerformanceList),
```
Comment (Maks19) [inline packages/web-app/src/modules/demand-monitor-views/DemandMonitorPage/DemandMonitorTable/DemandMonitorTable.tsx:147]: Why do you need `toJS` here?

File: packages/web-app/src/modules/demand-monitor-views/DemandMonitorPage/DemandMonitorTable/utils.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+export const sortHardwareDemandPerformance = (
+  demandedHardwarePerformanceList: DemandedHardwarePerformance[],
+  sortRule?: DemandMonitorTableSortRule,
+): DemandedHardwarePerformance[] => {
+  if (!sortRule) {
+    return demandedHardwarePerformanceList
+  }
+
+  return demandedHardwarePerformanceList.sort(sortRule)
+}
```
Comment (Maks19) [inline packages/web-app/src/modules/demand-monitor-views/DemandMonitorPage/DemandMonitorTable/utils.ts:25]:
```suggestion
export const sortHardwareDemandPerformance = (
  demandedHardwarePerformanceList: DemandedHardwarePerformance[],
  sortOrder: 'ascending' | 'descending' | 'none',
  sortRule?: DemandMonitorTableSortRule,
): DemandedHardwarePerformance[] => {
  if (!sortRule) {
    return demandedHardwarePerformanceList
  }

  const sortedList = demandedHardwarePerformanceList.sort(sortRule)
  return sortOrder === 'ascending' ? sortedList.reverse() : sortedList;
}
```

File: packages/web-app/src/modules/demand-monitor-views/DemandMonitorPage/DemandMonitorTable/DemandMonitorTable.tsx [hunk 3]
```diff
@@ -76,74 +101,135 @@ const styles: (theme: SaladTheme) => Record<string, CSS.Properties> = (theme: Sa
...
+    demandMonitorTableColumns[sortOrder.columnKey].sortRule,
+  )
+
+  if (sortOrder.sorted === 'ascending') {
+    sortedDemandedHardwarePerformanceList = sortedDemandedHardwarePerformanceList.reverse()
+  }
```
Comment (Maks19) [inline packages/web-app/src/modules/demand-monitor-views/DemandMonitorPage/DemandMonitorTable/DemandMonitorTable.tsx:153]:
```suggestion

  const sortedDemandedHardwarePerformanceList = sortHardwareDemandPerformance(
    toJS(demandedHardwarePerformanceList),
    sortOrder.sorted,
    demandMonitorTableColumns[sortOrder.columnKey].sortRule,
  )
```

**Review by Maks19 — CHANGES_REQUESTED** [review]

File: packages/web-app/src/modules/demand-monitor-views/DemandMonitorPage/DemandMonitorTable/DemandMonitorTable.tsx [hunk 4]
```diff
@@ -76,74 +101,135 @@ const styles: (theme: SaladTheme) => Record<string, CSS.Properties> = (theme: Sa
...
                   </Text>
-                </div>
-              </td>
-            </tr>
-          ))}
+                </td>
+                <td className={classes.tableCell}>
+                  <div className={classes.tableCellCentered}>
+                    <Text variant="baseXS">{recommendedSpecs.ramGb} GB System RAM</Text>
+                    {/* <Text variant="baseXS">{recommendedSpecs.storage}</Text> */}
```
[the remark this answers is not in the document]
Comment (vitto-moz) [reply packages/web-app/src/modules/demand-monitor-views/DemandMonitorPage/DemandMonitorTable/DemandMonitorTable.tsx:202]: we probably will make this value hardcoded, we will have a decision today

File: packages/web-app/src/modules/demand-monitor-views/DemandMonitorPage/DemandMonitorTable/DemandMonitorTable.tsx (hunk 1 above)
Comment (vitto-moz) [reply packages/web-app/src/modules/demand-monitor-views/DemandMonitorPage/DemandMonitorTable/DemandMonitorTable.tsx:146]: MobX  stores the data as a proxy object, so it doesn't have all the regular methods of `Array`, for example, that's why we need to transform it to related data type

File: packages/web-app/src/modules/demand-monitor-views/DemandMonitorPage/DemandMonitorPage.tsx [hunk 5]
```diff
@@ -50,9 +51,20 @@ const styles: (theme: SaladTheme) => Record<string, CSS.Properties> = (theme: Sa
...
+  demandedHardwarePerformanceList: DemandedHardwarePerformance[]
+}
+
+const _DemandMonitorPage: FunctionComponent<Props> = ({
+  classes,
+  fetchDemandedHardwarePerformanceList,
+  demandedHardwarePerformanceList,
+}) => {
+  useEffect(() => {
+    fetchDemandedHardwarePerformanceList()
```
[the remark this answers is not in the document]
Comment (vitto-moz) [reply packages/web-app/src/modules/demand-monitor-views/DemandMonitorPage/DemandMonitorPage.tsx]: yes, re-fetch once per hour functionality - added

File: packages/web-app/src/modules/demand-monitor-views/DemandMonitorPage/DemandMonitorTable/DemandMonitorTable.tsx (hunk 1 above)
Comment (vitto-moz) [reply packages/web-app/src/modules/demand-monitor-views/DemandMonitorPage/DemandMonitorTable/DemandMonitorTable.tsx]: `toJS` - removed, no longer needed

**Review by Maks19 — APPROVED** [review]

**Review by Maks19 — APPROVED** [review]

**Pull request merged**